grafana/provider.go

```diff
@@ -58,10 +68,46 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	} else {
 		cfg.APIKey = auth[0]
 	}
+
+	headersObj := d.Get("headers").(map[string]interface{})
+	if headersObj != nil && len(headersObj) == 0 {
```

Comment:
This can be simplified, as a `nil` map will always have a `len` of 0:
```suggestion
	if len(headersObj) == 0 {
```

Reply:
Simplified.

Reply:
Reverted, because it results in:

```
panic: interface conversion: interface {} is nil, not map[string]interface {} [recovered]
	panic: interface conversion: interface {} is nil, not map[string]interface {}
```